onnxruntime/core/providers/webnn/builders/impl/split_op_builder.cc

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Intel Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/safeint.h"
+#include "core/optimizer/initializer.h"
+#include "core/providers/common.h"
+#include "core/providers/shared/utils/utils.h"
+#include "core/providers/webnn/builders/helper.h"
+#include "core/providers/webnn/builders/model_builder.h"
+#include "core/providers/webnn/builders/op_builder_factory.h"
+
+#include "base_op_builder.h"
+
+namespace onnxruntime {
+namespace webnn {
+
+class SplitOpBuilder : public BaseOpBuilder {
+  // Add operator related.
+ private:
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                               const logging::Logger& logger) const override ORT_MUST_USE_RESULT;
+
+  // Operator support related.
+ private:
+  bool IsOpSupportedImpl(const InitializedTensorSet& initializers, const Node& node,
+                         const logging::Logger& logger) const override;
+
+  int GetMinSupportedOpSet(const Node& node) const override;
+};
+
+Status SplitOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
+                                             const Node& node,
+                                             const logging::Logger& logger) const {
+  const auto& input_defs = node.InputDefs();
+  emscripten::val input = model_builder.GetOperand(input_defs[0]->Name());
+  emscripten::val output_array;
+  std::vector<int64_t> input_shape;
+  ORT_RETURN_IF_NOT(GetShape(*input_defs[0], input_shape, logger), "Cannot get shape");
+  const auto rank = input_shape.size();
+  emscripten::val options = emscripten::val::object();
+
+  NodeAttrHelper helper(node);
+  int64_t axis = helper.Get("axis", 0);
+  axis = HandleNegativeAxis(axis, rank);
+  options.set("axis", static_cast<int32_t>(axis));
+
+  if (input_defs.size() == 2) {
+    // Inputs contains optional 'split' input
+    std::vector<int64_t> split;
+    const auto& initializers(model_builder.GetInitializerTensors());
+    const auto& split_tensor = *initializers.at(input_defs[1]->Name());
+    ORT_RETURN_IF_NOT(GetShapeByTensor(split_tensor, split, logger), "Cannot get split.");
+    std::vector<int32_t> splits;
+    std::transform(split.cbegin(), split.cend(),
+                   std::back_inserter(splits),
+                   [](int64_t dim) -> int32_t { return SafeInt<int32_t>(dim); });
+    output_array = model_builder.GetBuilder().call<emscripten::val>("split", input, emscripten::val::array(splits), options);
+    ORT_RETURN_IF_NOT(output_array["length"].as<int32_t>() == static_cast<int32_t>(split.size()), "The size of outputs must be equal to the size of 'split' input.");
+  } else {
+    if (helper.HasAttr("num_outputs")) {
+      const int64_t num_outputs = helper.Get("num_outputs", 1);
+      if (input_shape[axis] % num_outputs == 0) {
```

Comment:
Potential division by zero. We should verify `num_outputs > 0` and `ORT_RETURN_IF_NOT` if not.